server_addon/maya/client/ayon_maya/plugins/publish/collect_yeti_rig.py

```diff
@@ -271,22 +272,25 @@ def get_sequence(self, filepath, pattern="%04d"):
             pattern (str): The pattern to swap with the variable frame number.
 
         Returns:
-            list: file sequence.
+            list[str]: file sequence.
 
         """
-        import clique
-
-        escaped = re.escape(filepath)
-        re_pattern = escaped.replace(pattern, "-?[0-9]+")
-
+        filename = os.path.basename(filepath)
+        re_pattern = re.escape(filename)
+        re_pattern = re_pattern.replace(re.escape(pattern), "-?[0-9]+")
         source_dir = os.path.dirname(filepath)
-        files = [f for f in os.listdir(source_dir)
-                 if re.match(re_pattern, f)]
-
-        pattern = [clique.PATTERNS["frames"]]
-        collection, remainder = clique.assemble(files, patterns=pattern)
-
-        return collection
+        files = [f for f in os.listdir(source_dir) if re.match(re_pattern, f)]
+        collections, _remainder = clique.assemble(
+            files,
+            patterns=[clique.PATTERNS["frames"]],
+            minimum_items=1)
+            
+        if len(collections) > 1:
+            raise ValueError(
+                f"Multiple collections found for {collections}. "
+                "This is a bug.")
+
+        return list(collections[0])
```

Comment:
Had another look at the code. We have two issues.

1. This function MUST return nothing if there's no collection found, because it may be called for multiple root paths if the original texture path was relative and we then search across the image search paths.

2. We must return the full filepaths, not the filenames.

```suggestion
        files = [f for f in os.listdir(source_dir) if re.match(re_pattern, f)]
        if not files:
            # Files do not exist, this may not be a problem if e.g. the
            # textures were relative paths and we're searching across
            # multiple image search paths.
            return
        
        collections, _remainder = clique.assemble(
            files,
            patterns=[clique.PATTERNS["frames"]],
            minimum_items=1)
            
        if len(collections) > 1:
            raise ValueError(
                f"Multiple collections found for {collections}. "
                "This is a bug.")

        return [
            os.path.join(source_dir, filename) 
            for filename in collections[0]
        ]
```